kitsune/src/state.rs

```diff
@@ -109,51 +97,28 @@ impl SessionConfig {
 }
 
 impl_from_ref! {
-    Zustand;
+    AppState;
     [
-        cookie::Key => |input: &Zustand| input.session_config.cookie_key.clone(),
-        axum_flash::Config => |input: &Zustand| input.session_config.flash_config.clone()
+        cookie::Key => |input: &AppState| input.session_config.cookie_key.clone(),
+        axum_flash::Config => |input: &AppState| input.session_config.flash_config.clone()
     ]
 }
 
-/// Service collection
-///
-/// This contains all the "services" that Kitsune consists of.
-/// These are things like the search service, post service, etc.
-#[derive(Clone)]
-pub struct Service {
-    pub account: AccountService,
-    pub attachment: AttachmentService,
-    pub captcha: CaptchaService,
-    pub federation_filter: FederationFilterService,
-    pub job: JobService,
-    pub mailing: MailingService,
-    pub notification: NotificationService,
+#[derive(Clone, FromRef)]
+pub struct AppState {
+    pub core: CoreState,
     pub oauth2: OAuth2Service,
+    pub oauth_endpoint: OAuthEndpoint,
     #[cfg(feature = "oidc")]
     pub oidc: Option<OidcService>,
-    pub post: PostService,
-    pub instance: InstanceService,
-    pub search: SearchService,
-    pub timeline: TimelineService,
-    pub url: UrlService,
-    pub user: UserService,
+    pub session_config: SessionConfig,
 }
 
-/// Application state
-///
-/// Called it "Zustand" to avoid a name collission with `axum::extract::State`.
-/// "Zustand" is just the german word for state.
-#[derive(Clone, FromRef)]
-pub struct Zustand {
-    pub db_pool: PgPool,
-    pub embed_client: Option<EmbedClient>,
-    pub event_emitter: EventEmitter,
-    pub fetcher: Fetcher,
-    #[cfg(feature = "mastodon-api")]
-    pub mastodon_mapper: crate::mapping::MastodonMapper,
-    pub oauth_endpoint: OAuthEndpoint,
-    pub service: Service,
-    pub session_config: SessionConfig,
-    pub webfinger: Webfinger,
+// BAD BAD ANTI-PATTERN BAD
+impl std::ops::Deref for AppState {
```

Comment:
Not really a "good" Rust core, to be honest. This is basically emulating inheritance